saleor/dashboard/discount/views.py

```diff
@@ -75,11 +87,45 @@ def voucher_list(request):
 
 @staff_member_required
 @permission_required('discount.edit_voucher')
-def voucher_edit(request, pk=None):
-    if pk is not None:
-        instance = get_object_or_404(Voucher, pk=pk)
-    else:
-        instance = Voucher()
+def voucher_add(request):
+    instance = Voucher()
+    voucher_form = forms.VoucherForm(request.POST or None, instance=instance)
+    type_base_forms = {
```

Comment:
Since we're revisiting this code, could we convert this poor man's `switch` to a chain of `if`/`elif` statements? Currently we instantiate all of the possible forms which could result in unwanted side effects like database queries being evaluated even though in the end we will only ever need one of them.

Reply:
Sure, I'll change it.

Reply:
🎉

Reply:
It may require some adjustment to the logic as the template assumes it has all of them available so I'm not sure how viable this is.